### PR TITLE
[GOBBLIN-305] Add csv-kafka kafka-hdfs job template

### PR DESCRIPTION
**Pull request by zxcware**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/GOBBLIN-305

### Description
- [x] Here are some details about my PR:
  - Add 2 job templates: csv to kafka and kafka to hdfs
  - Add type transformation in `CsvToJsonConverterV2` converter

### Tests
- [x] My PR adds the following unit tests:
  - Changes are already covered by test in `CsvToJsonConverterV2Test`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by zxcware** [comment]
@htran1 Please review

File: gobblin-core/src/main/java/org/apache/gobblin/source/extractor/filebased/FileBasedSource.java [hunk 1]
```diff
@@ -215,10 +215,11 @@ protected void initLogger(SourceState state) {
       results = this.fsHelper.ls(path);
       for (int i = 0; i < results.size(); i++) {
         URI uri = new URI(results.get(i));
-        File file = uri.isAbsolute()?
-            new File(uri) : new File(state.getProp(ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY), uri.toString());
-
-        String filePath = file.getAbsolutePath();
+        String filePath = uri.toString();
```
Comment (htran1) [inline gobblin-core/src/main/java/org/apache/gobblin/source/extractor/filebased/FileBasedSource.java]: What's the reason for this change?

Comment (zxcware) [reply gobblin-core/src/main/java/org/apache/gobblin/source/extractor/filebased/FileBasedSource.java]: The previous implementation has an bug. It doesn't work for absolute uri with hdfs schema because java `File()` only represents local file, expecting schema `file://`
